### PR TITLE
Update CHANGELOG.md for 1.17.2 1.16.6 and 1.15.12

### PR DESCRIPTION
**Pull request by ldilalla-HC**
### Description
What does this PR do?

### TODO only if you're a HashiCorp employee
- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Review by fairclothjm — APPROVED** [review]
LGTM for auth/jwt. Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,31 @@
...
+### July 10, 2024
+
+CHANGES:
+
+* core: Bump Go version to 1.22.5
+* secrets/azure: Update plugin to v0.19.2 [[GH-27652](https://github.com/hashicorp/vault/pull/27652)]
+
+FEATURES:
+
+* **AWS secrets engine STS session tags support**: Adds support for setting STS
```
Comment (mladlow) [inline CHANGELOG.md:15]: We don't typically add "New Features" in minor releases so this one looks a little weird to me.

Comment (fairclothjm) [reply CHANGELOG.md:15]: I see this went into 1.17.2 https://github.com/hashicorp/vault/pull/27620

File: CHANGELOG.md [hunk 2]
```diff
@@ -227,6 +252,24 @@ autopilot to fail to discover new server versions and so not trigger an upgrade.
 * ui: fixed a bug where the replication pages did not update display when navigating between DR and performance [[GH-26325](https://github.com/hashicorp/vault/pull/26325)]
 * ui: fixes undefined start time in filename for downloaded client count attribution csv [[GH-26485](https://github.com/hashicorp/vault/pull/26485)]
 
+## 1.16.6 Enterprise
+### July 10, 2024
+
```
Comment (mladlow) [inline CHANGELOG.md:257]:
```suggestion
**Enterprise LTS:** Vault Enterprise 1.16 is a [Long-Term Support (LTS)](https://developer.hashicorp.com/vault/docs/enterprise/lts) release.

```

**Review by mladlow — COMMENTED** [review]
Can you add this one line?

File: CHANGELOG.md (hunk 1 above)
Comment (fairclothjm) [reply CHANGELOG.md:15]: I think we should not have backported the feature

**Review by mladlow — APPROVED** [review]

**Pull request merged**